repository-azure/src/main/java/io/aiven/elasticsearch/repositories/azure/AzureClient.java

```diff
@@ -57,18 +57,18 @@ public BlobServiceClient blobServiceClient() {
         return blobServiceClient;
     }
 
-    public static AzureClient create(final AzureStorageSettings azureStorageSettings) {
-        final var httpThreadPoolSettings = azureStorageSettings.httpThreadPoolSettings();
+    public static AzureClient create(final AzureKeystoreSettings azureKeystoreSettings) {
+        final var httpThreadPoolSettings = azureKeystoreSettings.httpThreadPoolSettings();
```

Comment:
Does not look like `httpThreadPoolSettings` belongs to `azureKeystoreSettings`, use `AzureSettings`?